tests/plugins/find_and_replace.test.ts

```diff
@@ -15,85 +15,252 @@ let searchOptions: SearchOptions;
 
 describe("basic search", () => {
   beforeEach(() => {
-    model = new Model();
+    model = new Model({ sheets: [{ id: "Sheet1" }] });
     setCellContent(model, "A1", "hello");
     setCellContent(model, "A2", "hello1");
     setCellContent(model, "A3", "=1");
     setCellContent(model, "A4", "111");
-    setCellContent(model, "A5", "1");
-    setCellContent(model, "A6", "2");
+    const sheet2 = "Sheet2";
+    createSheet(model, { activate: true, sheetId: sheet2 });
+    setCellContent(model, "A1", "hey");
+    setCellContent(model, "A2", "hey1");
+    setCellContent(model, "A3", "=2");
+    setCellContent(model, "A4", "222");
+    activateSheet(model, "Sheet1");
     searchOptions = {
       matchCase: false,
       exactMatch: false,
       searchFormulas: false,
+      searchScope: "thisSheet",
+      specificRange: undefined,
     };
   });
-  test("simple search", () => {
-    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
-    const matches = model.getters.getSearchMatches();
-    const matchIndex = model.getters.getCurrentSelectedMatchIndex();
-    expect(matches).toHaveLength(4);
-    expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 2, selected: false });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
-    expect(matches[3]).toStrictEqual({ col: 0, row: 4, selected: false });
-  });
+  test.each(["allSheets", "thisSheet", "specificRange"])(
```

Comment:
why would you ever do a test.each if there is a big `if/else` in the test wich next to nothing in common for all the cases ?